script.js

```diff
@@ -1,4 +1,398 @@
+
+  if (gameMode == "deal") {
+    allPlayersHand = dealCards(playerName.length);
+    computerCard = dealCards(1);
+    gameMode = "player turn";
+    return `Click "submit" for each player to start looking at their own hand.`;
+  }
+
+  if (gameMode == "player turn" && playerTurn < playerName.length) {
+    var playerCurrentTotal = calculateHand(allPlayersHand);
+    var computerCurrentTotal = calculateHand(computerCard);
+    gameMode = "player choice";
+    return `${playerName[playerTurn]}, you drew: <br> ${printHand(
+      allPlayersHand[playerTurn]
+    )} <br><br> Please input "hit" or "stand". `;
+  }
+
+  if (gameMode == "player choice") {
+    if (!(input == "hit" || input == "stand")) {
+      return `Please input only "hit" or "stand".`;
+    }
+
+    var output = "";
+    var lastPlayerDecided = false;
+    if (playerTurn == playerName.length - 1) {
+      lastPlayerDecided = true;
+      var lastStatement =
+        "All players have made their moves. <br><br> The dealer will make his move now.";
+    } else {
+      lastStatement = `Click "submit" for ${
+        playerName[playerTurn + 1]
+      }'s turn.`;
+    }
+
+    if (input == "hit") {
+      drawAdditionalCard(allPlayersHand[playerTurn]);
+      if (
+        calculateHand(allPlayersHand)[playerTurn] > 21 &&
+        lastPlayerDecided == false
+      ) {
+        gameMode = "player turn";
+        output = `On your hand: <br> ${printHand(
+          allPlayersHand[playerTurn]
+        )} <br><br> Your total has exceeded 21. You can only stand. <br><br> ${lastStatement}`;
+        playerTurn += 1;
+        return output;
+      } else if (
+        calculateHand(allPlayersHand)[playerTurn] > 21 &&
+        lastPlayerDecided
+      ) {
+        gameMode = "computer choice";
+        output = `On your hand: <br> ${printHand(
+          allPlayersHand[playerTurn]
+        )} <br><br> Your total has exceeded 21. You can only stand. <br><br> ${lastStatement}`;
+        playerTurn += 1;
+        return output;
+      } else {
+        return `On your hand: <br> ${printHand(
+          allPlayersHand[playerTurn]
+        )} <br><br> Please input "hit" or "stand".`;
+      }
+    }
+
+    if (input == "stand" && !lastPlayerDecided) {
+      gameMode = "player turn";
+      playerTurn += 1;
+      return `${lastStatement}`;
+    } else if (input == "stand" && lastPlayerDecided) {
+      gameMode = "computer choice";
+      playerTurn += 1;
+      return `${lastStatement}`;
+    }
+  }
+
+  if (gameMode == "computer choice") {
+    gameMode = "compare hands";
+    return computerDecisions(computerCard);
+  }
+
+  if (gameMode == "compare hands") {
+    var dealerOutput = `Dealer hand total is ${
+      calculateHand(computerCard)[0]
+    } points. <br><br> Dealer cards: <br>${printHand(computerCard[0])}`;
+    var finalOutput = compareHands(allPlayersHand, computerCard).join("");
+    resetGame();
+    return `${finalOutput} ${dealerOutput} <br><br> ${showLeaderboard()}`;
+  }
 };
```

Comment:
Great work on the helper functions above. The main is clean and easier to follow :) 

Looks like you've mastered the basics!  (loops, conditionals, global state) Looking forward to you leveling up from here!